api/test/wfl/integration/slack_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns wfl.integration.slack-test
+  (:require [clojure.test :refer :all]
+            [wfl.service.slack :as slack]
+            [wfl.util :as util])
+  (:import [clojure.lang PersistentQueue]))
+
+(def ^:private testing-agent (agent (PersistentQueue/EMPTY)))
+(def ^:private testing-slack-channel "C026PTM4XPA")
+(defn ^:private testing-slack-notification []
+  {:channel testing-slack-channel :message (format "WFL Integration Test Message: %s" (util/utc-now))})
+(def ^:private notify-promise (promise))
+(defn ^:private mock-send-notification
+  [queue]
+  (if (seq queue)
+    (let [{:keys [channel message]} (peek queue)
+          callback #(if (get % "ok")
+                      (deliver notify-promise true)
+                      (deliver notify-promise false))]
+      (slack/post-message channel message callback)
+      (pop queue))
+    queue))
+
+(add-watch testing-agent :watcher
+           (fn [_key _ref _old-state new-state]
+             (when (seq new-state)
+               (testing "notification is added to agent"
+                 (is (= (:channel (first (seq new-state))) testing-slack-channel))))))
+
+(deftest test-send-notification-to-a-slack-channel
+  (with-redefs-fn {#'slack/send-notification mock-send-notification}
+    #(do (slack/add-notification testing-agent (testing-slack-notification))
+         (send-off testing-agent #'slack/send-notification)
+         (testing "notification can actually be sent to slack"
+           (is (true? @notify-promise) "Slack notification promise should be fulfilled")))))
```

Comment:
Add a newline to the end of file.
Use `with-redefs` and lose the `#`?
I believe the promise works by blocking the calling thread on the agent's thread.

Reply:
It's super weird that once i switch to `with-redefs`, the test will timeout, even though the notification is sent.

Reply:
OK. We'll figure it out later.